File: tests/csharp/tests-csharp-xunit-api_request-playwright/Api/MockScenarios.cs
using Allure.Net.Commons.Attributes;

namespace Api;

/// <summary>
/// WireMock scenario switch for the mock stand (compose profile <c>mock</c>; the gateway
/// proxies <c>/__admin/</c> to WireMock).
/// </summary>
public static class MockScenarios
{
    public static bool Available()
    {
        try
        {
            var response = PlaywrightHttp.Request("GET", "/__admin/scenarios");
            return response.Status == 200;
        }
        catch (Exception)
        {
            return false;
        }
    }

    [AllureStep("Mock: switch scenario {scenario} to state {state}")]
    public static void SetState(string scenario, string state)
    {
        var response = PlaywrightHttp.Request(
            "PUT",
            $"/__admin/scenarios/{scenario}/state",
            json: new { state });
        Assert.True(response.Status == 200, response.Body);
    }

    [AllureStep("Mock: reset all scenarios")]
    public static void ResetAll()
    {
        var response = PlaywrightHttp.Request("POST", "/__admin/scenarios/reset");
        Assert.True(response.Status == 200, response.Body);
    }
}

File: tests/kotlin/tests-kotlin-junit5-ktor-selenium/src/test/kotlin/api/MockScenarios.kt
package api

import io.ktor.http.HttpMethod
import io.qameta.allure.Step

/**
 * WireMock scenario switch for the mock stand (compose profile `mock`; the gateway
 * proxies `/__admin/` to WireMock). Lets UI tests inject API failures that a healthy
 * live backend can never produce.
 *
 * In-cell HTTP is Ktor (same client as the api layer) — not a second folder.
 */
object MockScenarios {

    /** True when the stand under test exposes the WireMock admin API (mock stand only). */
    @JvmStatic
    fun available(): Boolean {
        return try {
            KtorHttp.request(HttpMethod.Get, "/__admin/scenarios").status == 200
        } catch (_: RuntimeException) {
            false
        }
    }

    @JvmStatic
    @Step("Mock: switch scenario {scenario} to state {state}")
    fun setState(scenario: String, state: String) {
        val response = KtorHttp.request(
            HttpMethod.Put,
            "/__admin/scenarios/$scenario/state",
            json = mapOf("state" to state),
        )
        check(response.status == 200) { response.body }
    }

    @JvmStatic
    @Step("Mock: reset all scenarios")
    fun resetAll() {
        val response = KtorHttp.request(HttpMethod.Post, "/__admin/scenarios/reset")
        check(response.status == 200) { response.body }
    }
}
